client/grpc_client.go

```diff
@@ -217,87 +240,115 @@ func (cli *grpcClient) finishAsyncCall(req *types.Request, res *types.Response)
 	return reqres
 }
 
+func (cli *grpcClient) checkErrGetResult() *types.Result {
```

Comment:
I'd just return types.Result directly, and check for res.IsOK() or res.IsErr(), just because pointers here are probably more expensive than the small types.Result structure.

And we can also do `return types.ErrInternalError.SetLog(errorLog)`.